Source/Dafny/Translator.cs

```diff
@@ -18735,9 +18735,19 @@ public class Substituter
           foreach (var bv in s.BoundVars) {
             substMap.Remove(bv);
           }
+
           // Put things together
           var rr = new ForallStmt(s.Tok, s.EndTok, newBoundVars, SubstAttributes(s.Attributes), Substitute(s.Range), s.Ens.ConvertAll(SubstMayBeFreeExpr), body);
           rr.Kind = s.Kind;
+          rr.CanConvert = s.CanConvert;
+          rr.Bounds = s.Bounds; // not substituted (should it be?)
```

Comment:
Yes, I think it should be (by calling `SubstituteBoundedPoolList`). The bounds may contain the variables that are being substituted.

Reply:
OK